ansible/roles/go-ipfs-deploy-docker/tasks/build.yml

```diff
@@ -0,0 +1,46 @@
+---
+- name: fetch GIT repo
+  become: no
+  run_once: true
+  git:
+    repo: "git@github.com:ipfs/go-ipfs.git"
+    force: yes
+    update: yes
+    dest: "/tmp/go-ipfs"
+    accept_hostkey: 'yes'
+    version: '{{ go_ipfs_version }}'
+  register: code_update
+
+- name: Build base image
+  become: no
+  run_once: true
+  docker_image:
+    name: "go-ipfs-ansible:{{ go_ipfs_version }}"
+    build:
+      path: /tmp/go-ipfs
+      pull: yes
+    source: build
+
+- name: Build final image with build args
+  become: no
+  run_once: true
+  docker_image:
+    name: "go-ipfs:{{ go_ipfs_version }}"
+    build:
+      path: "{{ role_path }}/files/dockerfiles/"
+      pull: no
+      args:
+        GO_IPFS_VERSION: "{{ go_ipfs_version }}"
+    source: build
+  register: docker_ipfs
+
+- name: Archive image
+  run_once: true
+  become: no
+  docker_image:
+    name: "go-ipfs"
+    tag: "{{ go_ipfs_version }}"
+    archive_path: "/tmp/go-ipfs-{{ go_ipfs_version }}.tar"
```

Comment:
The `archive` part should rather be part of `go-ipfs-deploy-docker/tasks/main.yml` as that is were it makes sense and is used. It is not obvious why the image would be archived after building and probably not every build needs to be archived (i.e. when deploying the binary).

Reply:
Sure, I'll move it but this would not run when updating the binary, it's for deploying the Docker image (bootstrappers only for now)